spi/core-spi/src/test/java/org/eclipse/dataspaceconnector/spi/types/domain/DataAddressTest.java

```diff
@@ -60,4 +58,48 @@ void verifyNullKeyThrowsException() {
                 .hasMessageContaining("Property key null.");
     }
 
+    @Test
+    void verifyCopy() {
+        DataAddress dataAddress = newSampleDataAddress();
+
+        var copy = dataAddress.copy();
+
+        assertSampleDataAddress(dataAddress, copy);
+    }
+
+    @Test
+    void verifyDeepCopy() {
+        DataAddress dataAddress = newSampleDataAddress();
+
+        var copy = dataAddress.copy();
+
+        var copyProperties = copy.getProperties();
+        copyProperties.put("foo", "new value");
+
+        assertEquals("bar", dataAddress.getProperty("foo"));
+    }
+
+    @Test
+    void verifyToBuilder() {
+        DataAddress dataAddress = newSampleDataAddress();
+
+        var copy = dataAddress.toBuilder().build();
+
+        assertSampleDataAddress(dataAddress, copy);
+    }
+
+    private DataAddress newSampleDataAddress() {
+        return DataAddress.Builder
+                .newInstance()
+                .type("test")
+                .keyName("somekey")
+                .property("foo", "bar")
+                .build();
+    }
+
+    private void assertSampleDataAddress(DataAddress dataAddress, DataAddress copy) {
```

Comment:
unclear from the method name what this method does. 

Using the proposed code you should be able to inline and remove the method anyway

Reply:
Correct, that method is obsolete now. I have used `usingRecursiveComparison()`.